sys/pm_callouts.c

```diff
@@ -408,7 +408,161 @@ void redir(portmaster_packet_info* packetInfo, portmaster_packet_info* redirInfo
     return;
 }
 
-void send_tcp_rst(portmaster_packet_info* packetInfo, void* originalPacket, ULONG originalPacketLength) {
+void send_icmp_blocked_packet(portmaster_packet_info* packetInfo, void* originalPacket, ULONG originalPacketLength, BOOL useLocalHost) {
+    // Only UDP is supported
+    if(packetInfo->protocol != 17) { // 17 -> UDP
+        return; // Not UDP
+    }
```

Comment:
I think we should use this for all protocols other than TCP - and possibly ICMP itself?.

Reply:
It send port unreachable. For ICMP it doesn't really make sens but probably will just be ignored so I don't think it is a problem.